src/extension.ts

```diff
@@ -198,6 +200,31 @@ export function activate(context: vscode.ExtensionContext): void {
                 node = target;
             }
 
+            if (vscode.workspace.workspaceFolders && vscode.workspace.workspaceFolders.length === 1) {
+                // only use the deployConfig is there is only one workspace opened
+                const deployConfig: { app: string, subpath: string } | undefined = vscode.workspace.getConfiguration(constants.extensionPrefix, vscode.workspace.workspaceFolders[0].uri).get(constants.configurationSettings.deploymentConfigurations);
+                if (deployConfig && deployConfig.app && deployConfig.subpath !== undefined) {
+                    const pathExists: boolean = await fs.pathExists(join(vscode.workspace.workspaceFolders[0].uri.fsPath, deployConfig.subpath));
+                    const nodeFromConfig: IAzureNode<WebAppTreeItem> | undefined = <IAzureNode<WebAppTreeItem>>await ext.tree.findNode(deployConfig.app); // resolves to undefined if app can't be found
+                    // tslint:disable-next-line:strict-boolean-expressions
+                    if (pathExists && nodeFromConfig) {
+                        // tslint:disable-next-line:strict-boolean-expressions
+                        if (!fsPath && (!node || node && node.id === nodeFromConfig.id)) {
+                            /*
+                            * only use the deployConfig in the following situation:
+                            * if there is no fsPath and no node, then the entry point was the deploy button
```

Comment:
> if there is a target node and it matches the id in the deployConfig

It feels like you should do the same thing, but for the fsPath as well (aka check to see if it matches fsPathFromConfig)